### PR TITLE
feat: create testimonial brand component in size small

### PR DESCRIPTION
**Pull request by rogervera**
## Testimonial Brand Component (small size)

This PR creates a testimonial component in the brand components package. Currently, this is just the small size (which does not have additional shapes as background styling). There is a light and dark version of the component.

---

Screen Shot 2020-02-24 at 11 58 10 AM

Screen Shot 2020-02-24 at 11 52 13 AM

## Merging your changes

When you are ready to merge to master, use the "squash and merge" button in GitHub, and follow the [commit message guide](/README.md#commit-message-guide) to write your commit message.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.2.4-alpha.dcb2d7.3+dcb2d7ca<br/>@codecademy/gamut-templates@2.2.35-alpha.dcb2d7.3+dcb2d7ca<br/>@codecademy/gamut@2.4.11-alpha.dcb2d7.3+dcb2d7ca<br/></pre>

File: packages/brand-components/src/Avatar/index.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ type AvatarImageProps =
     }
   | { alt?: never; 'aria-labelledby': string };
 
-export type AvatarBaseProps = {
```
Comment (JoshuaKGoldberg) [inline packages/brand-components/src/Avatar/index.tsx]: Unrelated change?

File: packages/brand-components/src/Testimonial/index.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <div
+      className={cx(s.contentContainer, s[`${size}Container`], {
+        [s.darkContainer]: theme === VisualTheme.DarkMode,
+        [s.lightContainer]: theme === VisualTheme.LightMode,
+      })}
+    >
+      <div className={s.avatarContainer}>
+        <Avatar
+          src={testimonial.imageUrl}
+          alt={`Photo of ${testimonial.name}`}
```
Comment (JoshuaKGoldberg) [inline packages/brand-components/src/Testimonial/index.tsx]: 👌

File: packages/brand-components/src/index.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
 export { default as Avatar } from './Avatar';
 export { default as Byline } from './Byline';
 export { default as Quote } from './Quote';
+export { default as Testimonial } from './Quote';
```
Comment (JoshuaKGoldberg) [inline packages/brand-components/src/index.ts]: ☝️

File: packages/brand-components/src/Testimonial/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+.smallContainer {
+  padding: 1.75rem 1.5rem 2rem;
+  width: calc(100% - 1rem);
+  max-width: px-rem(396px);
+  grid-template-columns: auto 1fr;
+  grid-gap: 1rem;
+  grid-template-areas:
+    "avatar byline"
+    "quote quote";
```
Comment (JoshuaKGoldberg) [inline packages/brand-components/src/Testimonial/styles.module.scss]: ~Are these used anywhere?~ well TIL

File: packages/brand-components/src/Testimonial/index.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  occupation: string;
+  imageUrl?: string;
+  quote: string;
+};
+
+type TestimonialProps = {
+  testimonial?: TestimonialContent;
+  size: 'small'; // TODO: 'medium' and 'large'
+  theme: VisualTheme;
+  bylineClassNames?: BylineClassNamesProps;
```
Comment (hmafzal) [inline packages/brand-components/src/Testimonial/index.tsx]: I'm confused by this prop - can you explain how to use it or what it does?

File: packages/brand-components/src/Avatar/index.tsx (hunk 1 above)
Comment (rogervera) [reply packages/brand-components/src/Avatar/index.tsx]: Yep.

File: packages/brand-components/src/index.ts (hunk 3 above)
Comment (JoshuaKGoldberg) [reply packages/brand-components/src/index.ts]: _(by nonblocking I suppose I mean nonblocking once fixed in this case 😛)_

File: packages/brand-components/src/Testimonial/index.tsx (hunk 5 above)
Comment (rogervera) [reply packages/brand-components/src/Testimonial/index.tsx]: Hmm, thanks for the question. You got me thinking.
Initially, I wanted to give the user of the component some flexibility on styling for the `name` and `occupation` `<span/>` tags the `<Byline/>` component renders.

But on closer thought, given that these components will be `small`,`medium`, or `large1 - that particular styling could be dictated by that size prop already just like the container.

Ex: `.smallByline`, `.mediumByline`, etc.

I feel comfortable removing.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.2.5-alpha.9ead35.5+9ead35c8<br/>@codecademy/gamut-templates@2.2.36-alpha.9ead35.5+9ead35c8<br/>@codecademy/gamut@2.4.12-alpha.9ead35.5+9ead35c8<br/></pre>

File: packages/brand-components/src/Testimonial/index.tsx (hunk 5 above)
Comment (hmafzal) [reply packages/brand-components/src/Testimonial/index.tsx]: Yeah, I feel like the flexibility for a component like this wouldn't be necessary since it's a brand component. It feels like components in this package should be a little more locked down in terms of styling than something in gamut.
